src/integrationTest/groovy/wooga/gradle/build/unity/tasks/UnityBuildPlayerTaskIntegrationSpec.groovy

```diff
@@ -57,6 +58,21 @@ class UnityBuildPlayerTaskIntegrationSpec extends UnityIntegrationSpec {
         !result.standardOutput.contains("toolsVersion=")
     }
 
+    @Issue("https://github.com/wooga/atlas-build-unity/issues/23")
+    def "clear buildTarget setting"() {
+        given: "a project setting default build target for all Unity tasks"
+        buildFile << """
+            unity.defaultBuildTarget = "ios"
+        """.stripIndent()
+
+        when:
+        def result = runTasksSuccessfully("exportCustom")
+
+        then:
+        result.standardOutput.contains("-buildTarget android")
+        !result.standardOutput.contains("-buildTarget ios")
```

Comment:
I actually didn't get it. Is it because `explortCustom` defaults to android somehow?

Reply:
https://github.com/wooga/atlas-build-unity/blob/20062445349ab30cd1f41e49e8134c218d0ae048/src/integrationTest/groovy/wooga/gradle/build/unity/tasks/UnityBuildPlayerTaskIntegrationSpec.groovy#L35

The test `appConfig` generated for all examples set the `buildTarget` to `android`.
This test sets the `defaultBuildTarget` to `ios`. I expect that we call **Unity** with the provided buildPlatform property from the `appConfig` and not the default value. But what actually happened in the bug was that we provided the `-buildTarget` flag multiple times (It speaks volumes that Unity allows this ... ;) ).

Reply:
Ok, got it, I've missed the `The test appConfig generated for all examples set the batchModeBuildTarget to android` part...